ng2-material/components/data_table/data_table_selectable_tr.ts

```diff
@@ -82,6 +82,8 @@ export class MdDataTableHeaderSelectableRow extends AbstractMdDataTableSelectabl
     this.table.onSelectableChange
       .map(event => event.allSelected)
       .subscribe(newActiveStatus => this.isActive = newActiveStatus);
+
+    this.onChange.subscribe(this.table.change.bind(this.table));
```

Comment:
I'm not an expert on this subject but I think too we should force the unsubscribe on destroy. I'm not sure it's done at the moment for subscription between checkbox and master checkbox.